resources/interface.go

```diff
@@ -11,3 +15,13 @@ type Filter interface {
 	Resource
 	Filter() error
 }
+
+var resourceListers = make(ResourceListers)
+
+func register(name string, lister ResourceLister) {
+	resourceListers[name] = lister
```

Comment:
would prevent adding the same twice here

Reply:
done